cpp/src/parquet/CMakeLists.txt

```diff
@@ -114,12 +114,13 @@ else()
   set(ARROW_LIBRARIES_FOR_STATIC_TESTS arrow_testing_shared arrow_shared)
 endif()
 
-set(PARQUET_MIN_TEST_LIBS ${ARROW_GTEST_GTEST_MAIN} ${ARROW_GTEST_GTEST} Boost::headers)
+set(PARQUET_MIN_TEST_LIBS ${ARROW_GTEST_GMOCK} ${ARROW_GTEST_GTEST}
```

Comment:
This patch add a `ARROW_GTEST_GMOCK` here? The latter `list(Append...` is just a style change?

Reply:
Yes. `ARROW_GTEST_GMOCK` is the main change. `list(APPEND` is for modenizing our CMake code.
It's not related to this fix. I'm modernizing our CMake code step-by-step when I touch related CMake code.
If you don't like it, I can remove the change from this PR.

Reply:
It's ok, just want to make it clear :-)